examples/thermostat/linux/thermostat-manager.cpp

```diff
@@ -0,0 +1,515 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**********************************************************
+ * Includes
+ *********************************************************/
+
+#include "include/thermostat-manager.h"
+#include "include/thermostat-delegate-impl.h"
+
+#include <app/clusters/bindings/BindingManager.h>
+#include <app/clusters/thermostat-server/thermostat-server.h>
+#include <controller/ReadInteraction.h>
+#include <platform/PlatformManager.h>
+
+/**********************************************************
+ * Defines and Constants
+ *********************************************************/
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::Controller;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Thermostat;
+using namespace chip::app::Clusters::Thermostat::Structs;
+using namespace chip::app::Clusters::Thermostat::Attributes;
+using namespace chip::app::Clusters::TemperatureMeasurement;
+using namespace chip::app::Clusters::TemperatureMeasurement::Attributes;
+using namespace Protocols::InteractionModel;
+
+using namespace chip::DeviceLayer;
+
+static constexpr EndpointId kThermostatEndpoint = 1;
+
+static constexpr uint16_t kMaxIntervalCeilingSeconds = 3600;
+
+static const char * SystemModeString(SystemModeEnum systemMode);
+static const char * RunningModeString(ThermostatRunningModeEnum runningMode);
+
+/**********************************************************
+ * Variable declarations
+ *********************************************************/
+
+ThermostatManager ThermostatManager::sThermostatMgr;
+
+namespace {
+
+CHIP_ERROR ChipErrorFromStatusCode(Status status)
```

Comment:
There's an existing `CHIP_ERROR_IM_GLOBAL_STATUS_VALUE` macro that does what you want here, I think.  Followup OK for switching to that.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/34591